src/number_theory/number_field/embeddings.lean

```diff
@@ -123,3 +124,141 @@ end
 end bounded
 
 end number_field.embeddings
+
+section place
+
+variables {A : Type*} [normed_division_ring A] {K : Type*} [field K] (φ : K →+* A)
+
+/-- An embedding into a normed division ring defines a place of `K` -/
+def number_field.place : absolute_value K ℝ :=
+{ to_fun := norm ∘ φ,
+  map_mul' := by simp only [function.comp_app, map_mul, norm_mul, eq_self_iff_true, forall_const],
+  nonneg' := by simp only [norm_nonneg, forall_const],
+  eq_zero' := by simp only [norm_eq_zero, map_eq_zero, forall_const, iff_self],
+  add_le' := by simp only [function.comp_app, map_add, norm_add_le, forall_const], }
+
+end place
+
+namespace number_field.complex_embeddings
+
+open complex number_field
+
+open_locale complex_conjugate
+
+variables {K : Type*} [field K]
+
+/-- The conjugate of a complex embedding as a complex embedding. -/
+def conjugate (φ : K →+* ℂ) : K →+* ℂ := ring_hom.comp conj_ae.to_ring_equiv.to_ring_hom φ
+
+lemma conjugate_coe_eq (φ : K →+* ℂ) (x : K) : (conjugate φ) x = conj (φ x) := rfl
+
+lemma place_conjugate_eq_place (φ : K →+* ℂ) : place (conjugate φ) = place φ :=
+by { ext1, simp only [place, conjugate_coe_eq, absolute_value.coe_mk, mul_hom.coe_mk,
+  function.comp_app, norm_eq_abs, abs_conj] }
+
+-- by { ext1, simp only [place, norm_eq_abs, absolute_value.coe_mk, mul_hom.coe_mk,
+--  function.comp_app] }
+
+/-- A embedding into `ℂ` is real if it is fixed by complex conjugation. -/
+def is_real (φ : K →+* ℂ): Prop := conjugate φ = φ
+
+/-- A real embedding as a ring homomorphism from `K` to `ℝ` . -/
+def real_embedding {φ : K →+* ℂ} (hφ : is_real φ) : K →+* ℝ :=
```

Comment:
Can you choose a name that allows dot notation? For example, if I have `h : is_real φ`, it would be nice to write `h.embedding` to get the corresponding `K →+* ℝ`.

Reply:
Yes. it is a good idea. I am not use yet on how to best use the dot notation.